src/org/rascalmpl/library/lang/html/IO.java

```diff
@@ -63,22 +89,118 @@ public IValue readHTMLFile(ISourceLocation file) {
             throw RuntimeExceptionFactory.io(factory.string(e.getMessage()));
         }
     }
+
+    /**
+     * Produces a HTML string output from an HTMLElement AST.
+     * 
+     * Why go through all the trouble of building a DOM? The only reason is compliance.
+     * Escapes, encodings, etc. all are maintained by these classes from org.w3c.dom.
+     */
+    public IString writeHTMLString(IConstructor cons) {
+        try {
```

Comment:
can maybe share more code with the `writeHTMLFile` call?

Reply:
👍 thanks for the suggestion. Especially since this code will become twice as long when we add the right XHTML dtd and wrap internal nodes with proper header and body.